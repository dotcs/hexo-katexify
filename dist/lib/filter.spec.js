"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const filter_1 = require("./filter");
describe("filter", () => {
    it("should not do anything if content does not contain katex formula", () => {
        const content = `
# This is some content
<!-- ignore this comment -->
<h2 class="to-be-ignored">Please ignore me</h2>
Here we have do no nothing. I repeat: Nothing!
    `;
        expect(filter_1.filter({ content })).toEqual({ content });
    });
    it("should find inline formula and replace it", () => {
        const content = `
# This is some content
<!-- ignore this comment -->
<h2 class="to-be-ignored">Please ignore me</h2>
This is a formula: $E = mc^2$.
    `;
        const { content: responseContent } = filter_1.filter({ content });
        expect(responseContent).not.toEqual(content);
        expect(responseContent).toContain("This is some content");
        expect(responseContent).toContain("This is a formula");
        expect(responseContent).not.toContain("katex-display");
        expect(responseContent).not.toContain("$E = mc^2$");
        expect(responseContent).not.toContain("$");
        expect(responseContent).toMatchSnapshot();
    });
    it("should find display formula and replace it", () => {
        const content = `
# This is some content
<!-- ignore this comment -->
<h2 class="to-be-ignored">Please ignore me</h2>
This is a formula: $$E = mc^2$$.
    `;
        const { content: responseContent } = filter_1.filter({ content });
        expect(responseContent).not.toEqual(content);
        expect(responseContent).toContain("This is some content");
        expect(responseContent).toContain("This is a formula");
        expect(responseContent).toContain("katex-display");
        expect(responseContent).not.toContain("$$E = mc^2$$");
        expect(responseContent).not.toContain("$");
        expect(responseContent).toMatchSnapshot();
    });
});
