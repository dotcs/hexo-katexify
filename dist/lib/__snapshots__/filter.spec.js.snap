// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter should find display formula and replace it 1`] = `
"
# This is some content
<!-- ignore this comment -->
<h2 class=\\"to-be-ignored\\">Please ignore me</h2>
This is a formula: <span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mi>E</mi><mo>=</mo><mi>m</mi><msup><mi>c</mi><mn>2</mn></msup></mrow><annotation encoding=\\"application/x-tex\\">E = mc^2</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.68333em;vertical-align:0em;\\"></span><span class=\\"mord mathdefault\\" style=\\"margin-right:0.05764em;\\">E</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.8641079999999999em;vertical-align:0em;\\"></span><span class=\\"mord mathdefault\\">m</span><span class=\\"mord\\"><span class=\\"mord mathdefault\\">c</span><span class=\\"msupsub\\"><span class=\\"vlist-t\\"><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.8641079999999999em;\\"><span style=\\"top:-3.113em;margin-right:0.05em;\\"><span class=\\"pstrut\\" style=\\"height:2.7em;\\"></span><span class=\\"sizing reset-size6 size3 mtight\\"><span class=\\"mord mtight\\">2</span></span></span></span></span></span></span></span></span></span></span></span>.
    "
`;

exports[`filter should find inline formula and replace it 1`] = `
"
# This is some content
<!-- ignore this comment -->
<h2 class=\\"to-be-ignored\\">Please ignore me</h2>
This is a formula: <span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mi>E</mi><mo>=</mo><mi>m</mi><msup><mi>c</mi><mn>2</mn></msup></mrow><annotation encoding=\\"application/x-tex\\">E = mc^2</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.68333em;vertical-align:0em;\\"></span><span class=\\"mord mathdefault\\" style=\\"margin-right:0.05764em;\\">E</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.8141079999999999em;vertical-align:0em;\\"></span><span class=\\"mord mathdefault\\">m</span><span class=\\"mord\\"><span class=\\"mord mathdefault\\">c</span><span class=\\"msupsub\\"><span class=\\"vlist-t\\"><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.8141079999999999em;\\"><span style=\\"top:-3.063em;margin-right:0.05em;\\"><span class=\\"pstrut\\" style=\\"height:2.7em;\\"></span><span class=\\"sizing reset-size6 size3 mtight\\"><span class=\\"mord mtight\\">2</span></span></span></span></span></span></span></span></span></span></span>.
    "
`;
